File: Data-structure-Algorithm/NumberTheoryAlgo/Method/binary search.cpp
#include<bits/stdc++.h>
using namespace std;
int n,i,j,k,l,h,a[100],x,t,temp;

int Binarysearch(int a[],int l,int h)
{
    if(h>=l)
    {
        int mid=l+(h-l)/2;

        if(a[mid]==x)
        {
            t=mid;
            temp=t;
            Binarysearch(a,l,mid-1);
        }
        else if(a[mid]>x)
            Binarysearch(a,l,mid-1);
        else if(a[mid]<x)
            Binarysearch(a,mid+1,h);

    }
    else
        return temp;

}


int main()
{

    cin>>n;
    for(i=1; i<=n; i++)
        cin>>a[i];
    sort(a,a+n+1);
    cin>>x;
    cout<<Binarysearch(a,1,n);

}

File: C#JobQuery/C#NumTheoryAlgo-STL-DS-Algo/NumberTheoryAlgo/NumberTheoryAlgo/SearchingSortingModule/BinarySearch.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SearchingSortingModule
{
    class BinarySearch
    {
        private static int BinarySearch_function(List<int> list, int l, int h, int s)
        {
            if (l <= h)
            {
                int mid = l + (h - l) / 2;
                if (list[mid] == s)
                    return s;
                else if (list[mid] > s)
                    return BinarySearch_function(list, l, mid - 1, s);
                else
                   return BinarySearch_function(list, mid + 1, h, s);
            }
            else
                return -1;
        }
        public static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            var line = Console.ReadLine().Split(' ');
            var list = new List<int>();
            for(int i = 0; i < n; i++)
                list.Add(int.Parse(line[i]));
   
            list.Sort();
            int s = int.Parse(Console.ReadLine());
            int result = BinarySearch_function(list, 0, n - 1, s);
            if(result == s)
                Console.WriteLine("value found");
            else
                Console.WriteLine("value not found");
        }

        
    }
}
